ee/codegen/src/utils/paths.ts

```diff
@@ -37,11 +39,42 @@ export function getGeneratedNodesModulePath(
   return modulePath;
 }
 
-export function getGeneratedNodeModulePath(
-  workflowContext: WorkflowContext,
-  moduleName: string
-): string[] {
-  return [...getGeneratedNodesModulePath(workflowContext), moduleName];
+export function getGeneratedNodeModuleInfo({
+  workflowContext,
+  nodeDefinition,
+  nodeLabel,
+}: {
+  workflowContext: WorkflowContext;
+  nodeDefinition: WorkflowNodeDefinition | undefined;
+  nodeLabel: string;
+}): { moduleName: string; nodeClassName: string; modulePath: string[] } {
+  const modulePathLeaf =
+    nodeDefinition?.module?.[nodeDefinition.module.length - 1];
+
+  let moduleName: string;
+  let nodeClassName: string;
+
+  // In the case of adorned Nodes, we need to traverse the Adornment Node's definition to get
+  // info about the inner Node that it adorns.
+  if (modulePathLeaf && modulePathLeaf === "<adornment>") {
```

Comment:
[oos] In the future, we'll want to handle multiple adornments as well. can defer and ticket for now